plugin/audit_log_v2/log_writer.h

```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2022 Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA */
+
+#ifndef LOG_WRITER_V2_H_INCLUDED
+#define LOG_WRITER_V2_H_INCLUDED
+
+#include "mysql/plugin_audit.h"
+
+#include "my_dir.h"
```

Comment:
Try to avoid including heavy includes from headers - very often forward declarations should be enough